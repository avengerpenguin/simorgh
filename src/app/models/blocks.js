export const rawVideoType = 'rawVideo';
export const imageType = 'image';
export const videoType = 'video';
export const rawImageType = 'rawImage';

export const singleTextBlock = text => ({
  type: 'text',
  model: {
    blocks: [
      {
        type: 'paragraph',
        model: {
          text,
        },
      },
    ],
  },
});

export const textBlock = text => ({
  blocks: [singleTextBlock(text)],
});

export const blockContainingText = (type, text) => ({
  type,
  model: textBlock(text),
});

export const blockBase = (blockType, blockModel) => ({
  type: blockType,
  model: blockModel,
});

export const blockArrayModel = arrayOfBlocks => ({
  blocks: arrayOfBlocks,
});

export const rawVideoModel = (
  videoLocator,
  videoVersionId,
  videoKind,
  videoDuration,
) => ({
  locator: videoLocator,
  versionID: videoVersionId,
  kind: videoKind,
  duration: videoDuration,
});

export const rawImageModel = imageLocator => ({
  locator: imageLocator,
});

export const stdArrayModelBlock = (blockType, modelArray) =>
  blockBase(blockType, blockArrayModel(modelArray));

export const rawVideoBlock = model => blockBase(rawVideoType, model);

export const rawImageBlock = model => blockBase(rawImageType, model);

export const imageBlock = rawBlock => stdArrayModelBlock(imageType, [rawBlock]);

export const videoBlock = (rawVBlock, imgBlock) =>
  stdArrayModelBlock(videoType, [rawVBlock, imgBlock]);
