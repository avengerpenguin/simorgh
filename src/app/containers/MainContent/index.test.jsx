import React from 'react';
import { shouldShallowMatchSnapshot } from '../../helpers/tests/testHelpers';
import MainContentContainer from './index';
import { blockContainingText } from '../../models/blocks';

describe('MainContent', () => {
  const blocks = [
    blockContainingText('headline', 'This is a headline!'),
    blockContainingText('subheading', 'This is a subheading!'),
    {
      type: 'text',
      blockId: '2',
      model: {
        blocks: [
          {
            type: 'paragraph',
            blockId: '2-1',
            model: {
              text: 'This is some text content!',
            },
          },
        ],
      },
    },
    {
      type: 'test', // causes prop validation errors
      blockId: '3',
      model: {
        blocks: [
          {
            model: {
              text: 'This is some test content!',
            },
          },
        ],
      },
    },
    {
      blockId: '4',
      type: 'image',
      model: {
        blocks: [
          {
            blockId: '4-1',
            type: 'rawImage',
            model: {
              width: 640,
              height: 420,
              locator:
                '/cpsprodpb/E7DB/production/_101655395_paulineclayton.jpg',
              originCode: null,
              copyrightHolder: 'BBC',
            },
          },
          blockContainingText('altText', 'Pauline Clayton'),
          blockContainingText(
            'caption',
            'Former embroider Pauline Clayton described the gift as "lovely"',
          ),
        ],
      },
    },
  ];

  shouldShallowMatchSnapshot(
    'should render correctly',
    <MainContentContainer blocks={blocks} />,
  );
});
