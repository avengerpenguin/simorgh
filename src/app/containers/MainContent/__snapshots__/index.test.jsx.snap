// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent should render correctly 1`] = `
<MainContent>
  <HeadingsContainer
    blocks={
      Array [
        Object {
          "blockId": "t-1",
          "model": Object {
            "blocks": Array [
              Object {
                "blockId": "p-1",
                "model": Object {
                  "text": "This is a headline!",
                },
                "type": "paragraph",
              },
            ],
          },
          "type": "text",
        },
      ]
    }
    type="headline"
  />
  <HeadingsContainer
    blocks={
      Array [
        Object {
          "blockId": "t-1",
          "model": Object {
            "blocks": Array [
              Object {
                "blockId": "p-1",
                "model": Object {
                  "text": "This is a subheading!",
                },
                "type": "paragraph",
              },
            ],
          },
          "type": "text",
        },
      ]
    }
    type="subheading"
    typeOfPreviousBlock="headline"
  />
  <TextContainer
    blocks={
      Array [
        Object {
          "blockId": "2-1",
          "model": Object {
            "text": "This is some text content!",
          },
          "type": "paragraph",
        },
      ]
    }
    type="text"
    typeOfPreviousBlock="subheading"
  />
  <BlockString
    blocks={
      Array [
        Object {
          "model": Object {
            "text": "This is some test content!",
          },
        },
      ]
    }
    type="test"
    typeOfPreviousBlock="text"
  />
  <ImageContainer
    blocks={
      Array [
        Object {
          "blockId": "4-1",
          "model": Object {
            "copyrightHolder": "BBC",
            "height": 420,
            "locator": "/cpsprodpb/E7DB/production/_101655395_paulineclayton.jpg",
            "originCode": null,
            "width": 640,
          },
          "type": "rawImage",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "blockId": "t-1",
                "model": Object {
                  "blocks": Array [
                    Object {
                      "blockId": "p-1",
                      "model": Object {
                        "text": "Pauline Clayton",
                      },
                      "type": "paragraph",
                    },
                  ],
                },
                "type": "text",
              },
            ],
          },
          "type": "altText",
        },
        Object {
          "model": Object {
            "blocks": Array [
              Object {
                "blockId": "t-1",
                "model": Object {
                  "blocks": Array [
                    Object {
                      "blockId": "p-1",
                      "model": Object {
                        "text": "Former embroider Pauline Clayton described the gift as \\"lovely\\"",
                      },
                      "type": "paragraph",
                    },
                  ],
                },
                "type": "text",
              },
            ],
          },
          "type": "caption",
        },
      ]
    }
    type="image"
    typeOfPreviousBlock="test"
  />
</MainContent>
`;
